error.go

```diff
@@ -0,0 +1,49 @@
+package es
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+)
+
+type Error struct {
+	StatusCode int
+	Type       string
+	Reason     string
+}
+
+func (e Error) Error() string {
+	return fmt.Sprintf("Error: [%d] %s: %s",
+		e.StatusCode,
+		e.Type,
+		e.Reason,
+	)
+}
+
+func ExtractError(res *esapi.Response) error {
+	if res.IsError() {
```

Comment:
lets reduce nesting? we can return early here and reduce nesting `if !res.IsError() { return nil }`

Reply:
ok